### PR TITLE
feat: change field group id and delete editor layout

### PR DESCRIPTION
**Pull request by IBratWurst**
## Summary
Functions to change field group id and delete editor layout

### CODE REVIEW

File: src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      && this.isSameContentType(other)
+  }
+  endsGroup(): boolean {
+    return false
+  }
+  shouldSave(): boolean {
+    return true
+  }
+  shouldPublish(): boolean {
+    return false
```
Comment (anho) [inline src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts:28]: Are updates to editor interfaces immediate?

File: src/lib/migration-steps/action-creators.ts [hunk 2]
```diff
@@ -330,6 +343,23 @@ const actionCreators = {
...
+        callsite: {
+          file: callsite.getFileName(),
+          line: callsite.getLineNumber()
+        }
+      },
+      payload: {
+        contentTypeId,
+        fieldGroupId,
+        props: {
+          newFieldGroupId
```
Comment (anho) [inline src/lib/migration-steps/action-creators.ts:359]: What's the reason to put this into a props object?

**Review by anho — COMMENTED** [review]
Beautiful, just minor clarifications.

File: src/lib/migration-steps/action-creators.ts (hunk 2 above)
Comment (IBratWurst) [reply src/lib/migration-steps/action-creators.ts:359]: Just to not pollute the type `RawStepPayload`, similarly to how it's used in field renaming

File: src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts (hunk 1 above)
Comment (IBratWurst) [reply src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts:28]: What do you intend as immediate? As far as I understood, all intents changes within a group are applied together.

Comment (anho) [reply src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts:28]: Immediate = they don't need to be published. I'm not sure about the lifecycle of editor_interfaces so just want to clarify.

File: src/lib/migration-steps/action-creators.ts (hunk 2 above)
Comment (anho) [reply src/lib/migration-steps/action-creators.ts:359]: Renaming a field seems to be the outlier in this file and usually all the info is put into the payload (as long as the keys are fixed).

File: src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts (hunk 1 above)
Comment (IBratWurst) [reply src/lib/intent/editor-layout/editor-layout-change-field-group-id.ts:28]: Yep, no publication of editor interfaces

File: src/lib/migration-steps/action-creators.ts (hunk 2 above)
Comment (IBratWurst) [reply src/lib/migration-steps/action-creators.ts]: updated!

**Review by anho — APPROVED** [review]

**Pull request merged**